src/main/groovy/com/bmuschko/gradle/java2html/ConvertCodeTask.groovy

```diff
@@ -15,38 +15,48 @@
  */
 package com.bmuschko.gradle.java2html
 
-
 import org.gradle.api.DefaultTask
 import org.gradle.api.file.FileCollection
+import org.gradle.api.tasks.CacheableTask
+import org.gradle.api.tasks.Classpath
 import org.gradle.api.tasks.Input
 import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.Optional
 import org.gradle.api.tasks.OutputDirectory
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.PathSensitivity
 import org.gradle.api.tasks.TaskAction
 
 /**
  * Java2HTML files documentation generation task.
  *
  * @author Benjamin Muschko
  */
+@CacheableTask
 class ConvertCodeTask extends DefaultTask {
-    @InputFiles FileCollection classpath
+    @Classpath FileCollection classpath
```

Comment:
This also needs to be annotated with `@PathSensitive(PathSensitivity.RELATIVE)`.

Reply:
OK